src/FileFormats/MOF/MOF.jl

```diff
@@ -123,15 +140,19 @@ Keyword arguments are:
  - `differentiation_backend::MOI.Nonlinear.AbstractAutomaticDifferentiation = MOI.Nonlinear.SparseReverseMode()`:
    automatic differentiation backend to use when reading models with nonlinear
    constraints and objectives.
+ - `parse_as_nlpblock::Bool=true`: if `true` parse `"ScalarNonlinearFunction"`
+   into an `MOI.NLPBlock`. If `false`, `"ScalarNonlinearFunction"` are parsed as
+   `MOI.ScalarNonlinearFunction` functions.
 """
 function Model(;
     print_compact::Bool = false,
     warn::Bool = false,
     differentiation_backend::MOI.Nonlinear.AbstractAutomaticDifferentiation = MOI.Nonlinear.SparseReverseMode(),
+    parse_as_nlpblock::Bool = true,
```

Comment:
Once this PR is merged I'll open a separate PR to debate setting this to `false` by default.